dask/array/utils.py

```diff
@@ -297,6 +298,12 @@ def assert_eq(
         assert (
             a.shape == b.shape
         ), f"a and b have different shapes (a: {a.shape}, b: {b.shape})"
+        if check_type:
+            _a = a if a.shape else a.item()
```

Comment:
What objects do we need to call `.item()` for?

Reply:
Arrays with shape 0  are effectively the same as just the `.item()` of themselves.

Reply:
For example:

```python
da.assert_eq(np.array(12.33), 12.33)
```

This passes `assert_eq` on main, and I wanted it to pass with `check_type` as well.

Reply:
FWIW doing `[()]` should also work. This has the added benefit of being a no-op for arrays with some shape

```python
In [1]: import numpy as np

In [2]: np.array(12)[()]
Out[2]: 12

In [3]: np.array([12])[()]
Out[3]: array([12])

In [4]: np.array([11, 12])[()]
Out[4]: array([11, 12])

In [5]: np.array([[12]])[()]
Out[5]: array([[12]])
```

Reply:
Oh nice! I am happy to do whichever people think is cleaner. I am checking shape, so the functionality is the same either way.

Reply:
No strong feelings. Either approach seems fine 🙂